Dockerfile

```diff
@@ -1,74 +1,103 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
 
-# Copy fluxbox configuration
-COPY --chown=0:0 config/fluxbox /home/user/.fluxbox/init
 
-# Copy predefined configs
-COPY --chown=0:0 config/etc /etc/
+# Stage 1. Prepare JetBrains IDE with Projector.
+#   1. Downloads JetBrains IDE packaging by given downloadUrl build argument.
+#   2. If buildGradle build argument is set to false, then consumes built Projector assembly from the host.
+#       2.1 Otherwise starts Gradle build of Projector Server and Projector Client.
+#   3. Copies static files to the Projector assembly (entrypoint, launcher, configuration).
+FROM registry.access.redhat.com/ubi8-minimal:8.3-298 as projectorAssembly
+ENV PROJECTOR_DIR /projector
+ENV JAVA_HOME /usr/lib/jvm/java-11
+ARG downloadUrl
+ARG buildGradle
+ADD jetbrains-editor-images/static $PROJECTOR_DIR
+ADD projector-client $PROJECTOR_DIR/projector-client
+ADD projector-server $PROJECTOR_DIR/projector-server
+WORKDIR /downloads
+RUN microdnf install -y wget findutils tar gzip unzip java-11-openjdk-devel \
+    && wget -q $downloadUrl -O - | tar -xz \
+    && find . -maxdepth 1 -type d -name * -exec mv {} $PROJECTOR_DIR/ide \;
+WORKDIR $PROJECTOR_DIR/projector-server
+RUN if [ "$buildGradle" = "true" ]; then ./gradlew clean; else echo "Skipping gradle build"; fi \
+    && if [ "$buildGradle" = "true" ]; then ./gradlew --console=plain :projector-server:distZip; else echo "Skipping gradle build"; fi
+WORKDIR $PROJECTOR_DIR
+RUN cp $PROJECTOR_DIR/projector-server/projector-server/build/distributions/projector-server-1.0-SNAPSHOT.zip $PROJECTOR_DIR \
+    && unzip $PROJECTOR_DIR/projector-server-1.0-SNAPSHOT.zip \
+    && rm $PROJECTOR_DIR/projector-server-1.0-SNAPSHOT.zip \
+    && rm -rf $PROJECTOR_DIR/projector-client \
+    && rm -rf $PROJECTOR_DIR/projector-server \
+    && mv projector-server-1.0-SNAPSHOT $PROJECTOR_DIR/ide/projector-server \
+    && mv $PROJECTOR_DIR/ide-projector-launcher.sh $PROJECTOR_DIR/ide/bin \
+    && chmod 644 $PROJECTOR_DIR/ide/projector-server/lib/*
 
-# Copy sh scripts
-COPY --chown=0:0 scripts/*.sh /opt/
+# Stage 2. Build the main image with necessary environment for running Projector
+#   Doesn't require to be a desktop environment. Projector runs in headless mode.
+FROM registry.access.redhat.com/ubi8-minimal:8.3-298
+ARG downloadUrl
+ENV PROJECTOR_USER_NAME projector-user
+ENV PROJECTOR_DIR /projector
+ENV HOME /home/$PROJECTOR_USER_NAME
+ENV PROJECTOR_CONFIG_DIR $HOME/.config
+RUN set -ex \
+    && microdnf install -y yum \
```

Comment:
are you installing `yum` for module support? if you don't need that, you could save some disk space and just use microdnf to install everything.

Reply:
So it is enough to use microdnf all the time instead of yum?

Reply:
Unless you need features in yum or dnf, then yes, microdnf is good enough for installing rpms. dnf/yum provide support for MODULE installs. See also comments below about the amount of stuff that you end up installing when you request `yum`... lots of Python rpms are needed for yum, whereas microdnf is tiny and has no such reqs.

Reply:
Removed yum from the Dockerfile.